### DISCUSSION TITLE
[Not Extension] Option to indent empty lines when using block indent. [with usable fork]

### DISCUSSION THREAD

**Discussion by xaxazak** [microsoft/vscode-discussions#148] [Extension Development QnA] [answered]
**disclaimer**: I know it's in the wrong category, there seems to be nowhere appropriate to discuss an issue like this, so apologies if I'm using the extension development category.

---

Say you have selected code looking like:

```
foo

zug
```

and you press indent. Some people (and/or coding styles) prefer that there are no lines with _only_ whitespace, even when in an intended region of code, so they would prefer the result to be `method A`:

```
[INDENT]foo

[INDENT]zug
```

Others prefer indents to be consistent (between brackets etc) _regardless_ of whether lines are empty or not, so they would prefer `method B`:

```
[INDENT]foo
[INDENT]
[INDENT]zug
```

Method B is also simpler and, IMHO, more intuitive, as there's no "except for" - block indent just indents _all_ selected lines.
However, the current behavior is to use `method A` _only_.

I created a [branch](https://github.com/xaxazak/vscode/tree/mod-indent-empty-lines) that adds an option that allows you to select `method A or B` (default = A). The option is `Always Indent Empty Lines`. It's a very small change and I endeavored to add it cleanly and properly, and I added tests.

I feel this would be a good option to add to the codebase, it's a tiny change and provides the preferred functionality for many people and coding styles. I added this as an [issue](https://github.com/microsoft/vscode/issues/113288) a while ago, but I think it flew under the radar as I never discussed it elsewhere.

---

There is another option (which would require a bit more coding) where you could have 3 options. 

`method C` would be to indent empty lines if and only if they were _between_ non-empty lines in the selection. So indenting the selection:

```

foo

bar

```

would result in:

```

[INDENT]foo
[INDENT]
[INDENT]bar

```

and not

```
[INDENT]
[INDENT]foo
[INDENT]
[INDENT]bar
[INDENT]
```

That should be fairly easy to add, but I'm not sure if it's surplus to requirements.

--

Does this seem like a sensible option for VSCode to have?
Does this patch look reasonable?

**Comment by burkeholland** [comment, marked as answer]
This would be better discussed as an issue on the VS Code repo. Closing here but feel free to open there with this same context.
